crates/nu-command/src/filesystem/ls.rs

```diff
@@ -81,241 +94,69 @@ impl Command for Ls {
-                        }
-                        return None;
+        match input_pattern_arg {
+            None => Ok(ls_for_one_pattern(None, args, ctrl_c.clone(), cwd)?
+                .into_pipeline_data_with_metadata(
+                    PipelineMetadata {
+                        data_source: DataSource::Ls,
+                    },
+                    ctrl_c,
+                )),
+            Some(pattern) => {
+                let mut result_iters = vec![];
+                let mut errors = vec![];
+                for pat in pattern {
+                    match ls_for_one_pattern(Some(pat), args, ctrl_c.clone(), cwd.clone()) {
+                        Ok(it) => result_iters.push(it),
+                        Err(e) => errors.push(e),
                     }
+                }
 
-                    let display_name = if short_names {
-                        path.file_name().map(|os| os.to_string_lossy().to_string())
-                    } else if full_paths || absolute_path {
-                        Some(path.to_string_lossy().to_string())
-                    } else if let Some(prefix) = &prefix {
-                        if let Ok(remainder) = path.strip_prefix(prefix) {
-                            if directory {
-                                // When the path is the same as the cwd, path_diff should be "."
-                                let path_diff =
-                                    if let Some(path_diff_not_dot) = diff_paths(&path, &cwd) {
-                                        let path_diff_not_dot = path_diff_not_dot.to_string_lossy();
-                                        if path_diff_not_dot.is_empty() {
-                                            ".".to_string()
-                                        } else {
-                                            path_diff_not_dot.to_string()
-                                        }
-                                    } else {
-                                        path.to_string_lossy().to_string()
-                                    };
-
-                                Some(path_diff)
-                            } else {
-                                let new_prefix = if let Some(pfx) = diff_paths(prefix, &cwd) {
-                                    pfx
-                                } else {
-                                    prefix.to_path_buf()
-                                };
-
-                                Some(new_prefix.join(remainder).to_string_lossy().to_string())
-                            }
-                        } else {
-                            Some(path.to_string_lossy().to_string())
+                // if no results available, need to report all errors and return Err
+                // or else return Error along with items.
+                if result_iters.is_empty() {
+                    if !errors.is_empty() {
+                        let last_error = errors.pop().expect("Already check errors is not empty");
```

Comment:
Same edit here

Reply:
I don't think we need this again, because nushell will return error on partial error :)